### ISSUE TITLE
Bug: GDCprepare(query) fails adding metadata for TCGA-SKCM gene expression

### ISSUE THREAD

**Issue by nietsenan** [BioinformaticsFMRP/TCGAbiolinks#291]
skcm_gene <- GDCquery(project = "TCGA-SKCM", 
+                       data.category = "Gene expression",
+                       data.type = "Gene expression quantification",
+                       platform = "Illumina HiSeq",
+                       file.type = "results",
+                       legacy = TRUE)

> GDCdownload(skcm_gene,method = "api")
> skcm_gene_Prepare <- GDCprepare(skcm_gene, summarizedExperiment = TRUE)
  |========================================================| 100%Error in fix.by(by.y, y) : 'by' must specify a uniquely valid column

**Comment by nietsenan** [comment]
> sessionInfo()
R version 3.5.2 (2018-12-20)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.2

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] zh_CN.UTF-8/zh_CN.UTF-8/zh_CN.UTF-8/C/zh_CN.UTF-8/zh_CN.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils    
[7] datasets  methods   base     

other attached packages:
 [1] SummarizedExperiment_1.12.0 DelayedArray_0.8.0         
 [3] BiocParallel_1.16.5         matrixStats_0.54.0         
 [5] Biobase_2.42.0              GenomicRanges_1.34.0       
 [7] GenomeInfoDb_1.18.1         IRanges_2.16.0             
 [9] S4Vectors_0.20.1            BiocGenerics_0.28.0        
[11] TCGAbiolinks_2.11.2        

loaded via a namespace (and not attached):
  [1] colorspace_1.4-0            selectr_0.4-1              
  [3] rjson_0.2.20                hwriter_1.3.2              
  [5] circlize_0.4.5              XVector_0.22.0             
  [7] GlobalOptions_0.1.0         rstudioapi_0.9.0           
  [9] ggpubr_0.2                  matlab_1.0.2               
 [11] ggrepel_0.8.0               bit64_0.9-7                
 [13] AnnotationDbi_1.44.0        xml2_1.2.0                 
 [15] codetools_0.2-16            splines_3.5.2              
 [17] R.methodsS3_1.7.1           doParallel_1.0.14          
 [19] DESeq_1.34.1                geneplotter_1.60.0         
 [21] knitr_1.21                  jsonlite_1.6               
 [23] Rsamtools_1.34.1            km.ci_0.5-2                
 [25] broom_0.5.1                 annotate_1.60.0            
 [27] cluster_2.0.7-1             R.oo_1.22.0                
 [29] readr_1.3.1                 compiler_3.5.2             
 [31] httr_1.4.0                  backports_1.1.3            
 [33] assertthat_0.2.0            Matrix_1.2-15              
 [35] lazyeval_0.2.1              limma_3.38.3               
 [37] prettyunits_1.0.2           tools_3.5.2                
 [39] bindrcpp_0.2.2              gtable_0.2.0               
 [41] glue_1.3.0                  GenomeInfoDbData_1.2.0     
 [43] dplyr_0.7.8                 ggthemes_4.0.1             
 [45] ShortRead_1.40.0            Rcpp_1.0.0                 
 [47] Biostrings_2.50.2           nlme_3.1-137               
 [49] rtracklayer_1.42.1          iterators_1.0.10           
 [51] xfun_0.4                    stringr_1.3.1              
 [53] rvest_0.3.2                 XML_3.98-1.16              
 [55] edgeR_3.24.3                zoo_1.8-4                  
 [57] zlibbioc_1.28.0             scales_1.0.0               
 [59] aroma.light_3.12.0          hms_0.4.2                  
 [61] RColorBrewer_1.1-2          curl_3.3                   
 [63] ComplexHeatmap_1.20.0       yaml_2.2.0                 
 [65] memoise_1.1.0               gridExtra_2.3              
 [67] KMsurv_0.1-5                ggplot2_3.1.0              
 [69] downloader_0.4              biomaRt_2.38.0             
 [71] latticeExtra_0.6-28         stringi_1.2.4              
 [73] RSQLite_2.1.1               genefilter_1.64.0          
 [75] foreach_1.4.4               GenomicFeatures_1.34.3     
 [77] shape_1.4.4                 rlang_0.3.1                
 [79] pkgconfig_2.0.2             bitops_1.0-6               
 [81] lattice_0.20-38             purrr_0.3.0                
 [83] bindr_0.1.1                 cmprsk_2.2-7               
 [85] GenomicAlignments_1.18.1    bit_1.1-14                 
 [87] tidyselect_0.2.5            plyr_1.8.4                 
 [89] magrittr_1.5                R6_2.3.0                   
 [91] generics_0.0.2              DBI_1.0.0                  
 [93] mgcv_1.8-26                 pillar_1.3.1               
 [95] survival_2.43-3             RCurl_1.95-4.11            
 [97] tibble_2.0.1                EDASeq_2.16.3              
 [99] crayon_1.3.4                survMisc_0.5.5             
[101] GetoptLong_0.1.7            progress_1.2.0             
[103] locfit_1.5-9.1              grid_3.5.2                 
[105] sva_3.30.1                  data.table_1.12.0          
[107] blob_1.1.1                  ConsensusClusterPlus_1.46.0
[109] digest_0.6.18               xtable_1.8-3               
[111] tidyr_0.8.2                 R.utils_2.7.0              
[113] munsell_0.5.0               survminer_0.4.3

**Comment by nietsenan** [comment]
Hi, Tiago. I want to prepare the gene expression data of TCGA-SKCM with the GDCprepare(). However, it always fails with "Error in fix.by(by.y, y) : 'by' must specify a uniquely valid column".

**Comment by FiveThirteen** [comment]
Yes, I have the same problem. This just started happening a few days ago, scripts were running perfectly prior to that.

**Comment by mkarikom** [comment]
Same issue with GDCPrepare():
`Error in fix.by(by.y, y) : 'by' must specify a uniquely valid column`

**Comment by tiagochst** [comment]
Do you have installed TCGAbiolinks 2.11.3 ?

**Comment by nietsenan** [comment]
TCGAbiolinks 2.11.3 requires R(version 3.6), however, there is no version 3.6 R for macos now.  So we can not install the newest TCGAbiolinks...

**Comment by nietsenan** [comment]
I tried with the TCGAbiolinks_2.11.2 again. Surprisingly, the GDCprepare function works well now..... Thank you !!

**Comment by FiveThirteen** [comment]
Now it works for me again as well with TCGAbiolinks_2.11.2 on a Mac. Thank you Dr. Silva!

**Comment by mkarikom** [comment]
> Do you have installed TCGAbiolinks 2.11.3 ?

Yes, but it seems to be working again!  Thanks for your help.
